### PR TITLE
Xml rpc requests refactoring

### PR DESCRIPTION
**Pull request by oklinov**

### CODE REVIEW

File: koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/ArchiveList.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        this.archives = archives;
+    }
+
+    public ArchiveList(Object object) {
+        archives = parseArchiveMaps(toMaps(object));
+    }
+
+    @Override
+    public Object toObject() {
+        return new Object[]{};
```
Comment (sparkoo) [inline koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/ArchiveList.java]: surprising, now I'm even more confused what this `toObject` means

File: koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/BuildList.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    private Map<String, Object> parseBuild(Build build) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(Constants.name, build.getName());
+        map.put(Constants.version, build.getVersion());
+        map.put(Constants.release, build.getRelease());
+        map.put(Constants.nvr, build.getNvr());
+        map.put(Constants.build_id, build.getId());
+        map.put(Constants.completion_time, build.getCompletionTime());
+        // map.put(Constants.rpms, parseRpms(build.getRpms()));
```
Comment (sparkoo) [inline koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/BuildList.java]: why is this commented out? can we remove it?

File: koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/PackageId.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+public class PackageId implements XmlRpcResponse<Integer> {
+
+    private final Integer packageId;
+
+    public PackageId(Integer packageId) {
+        this.packageId = packageId;
+    }
+
+    public PackageId(Object object) {
```
Comment (sparkoo) [inline koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/PackageId.java]: Overall, I'm for removing all constructors accepting `Object` and create static factory for it.

File: koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/XmlRpcResponse.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package org.fakekoji.xmlrpc.server.xmlrpcresponse;
+
+public interface XmlRpcResponse <T> {
```
Comment (sparkoo) [inline koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/XmlRpcResponse.java]: again, please javadoc this.

File: koji-scm-lib/src/main/java/hudson/plugins/scm/koji/client/BuildMatcher.java [hunk 5]
```diff
@@ -114,39 +122,29 @@ public BuildMatcher(String currentURL, Predicate<String> notProcessedNvrPredicat
...
         }
-        final XmlRpcRequestParams params = new XmlRpcRequestParamsBuilder()
-                .setPackageId(packageId)
-                .setStarstar(Boolean.TRUE)
-                .setState(1)
-                .build();
-        final XmlRpcResponse response = execute(Constants.listBuilds, params);
-        List<Build> builds = response.getBuilds();
+        final XmlRpcRequestParams params = new ListBuilds(packageId);
+        final BuildList response = (BuildList) execute(params);
```
[the remark this answers is not in the document]
Comment (oklinov) [reply koji-scm-lib/src/main/java/hudson/plugins/scm/koji/client/BuildMatcher.java]: Didn't find better solution. Do you insist on refactoring?

File: koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/ArchiveList.java (hunk 1 above)
Comment (oklinov) [reply koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/ArchiveList.java]: Normally, it shoud return array of maps (as array of objects), but it is not implemented in fake-koji so I return empty array

File: koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/BuildList.java (hunk 2 above)
Comment (oklinov) [reply koji-scm-lib/src/main/java/org/fakekoji/xmlrpc/server/xmlrpcresponse/BuildList.java]: I forgot to remove it. RPMs are retrieved in getRPMs method, not in listBuilds

**Pull request merged**